src/poller.rs

```diff
@@ -339,91 +339,51 @@ impl JobPoller {
         let monitor_task_name = format!("job-{}-monitor-{}", job_type, job_id);
 
         spawn_named_task!(&monitor_task_name, async move {
+            use tracing::Instrument;
+
             tokio::pin!(job_handle);
 
             tokio::select! {
-                // Biased select ensures we check for shutdown signal first
-                biased;
-
+                _ = &mut job_handle => {
+                    // Job completed - no need for shutdown coordination
+                }
                 Ok(shutdown_notifier) = shutdown_rx.recv() => {
-                    // Create a span for this specific job's shutdown sequence
-                    let shutdown_span = tracing::info_span!(
-                        parent: None,
-                        "job.shutdown_coordination",
-                        job_id = %job_id,
-                        job_type = %job_type,
-                        coordination_path = "shutdown_first",
-                        ack_sent = tracing::field::Empty,
-                        job_completed = tracing::field::Empty,
-                    );
-
-                    let _enter = shutdown_span.enter();
```

Comment:
`enter()` doesn't work across `.await` points. That's why the `Instrument` trait exists - to work for futures.